### ISSUE TITLE
Exception when RemoveAtOnScheduler on ReactiveCollection

### ISSUE THREAD

**Issue by nomadev** [runceel/ReactiveProperty#274]
Hello,
I am experiencing a strange behaviour in my WPF application.

Consider this command (using DevExpress POCO viewmodel command): 
```
public async Task RemoveAtIndex()
        {
            try
            {
                var res = await askToSaveUnsavedChanges(MyReactiveCollection.ElementAt(SelectedIndex));
                if (res == MessageBoxResult.Cancel) return;
                log.Debug($"removing index {SelectedIndex}");
                MyReactiveCollection.RemoveAtOnScheduler(SelectedIndex); // exception occurs after this line is executed
            }
            catch (Exception e)
            {
                logError(e);
            }
        }
```
Sometimes I get this exception:
Exception has been thrown by the target of an invocation.
Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index

I do log the specified index (SelectedIndex), and I am sure it is never negative and it is always correct. 

I was using the ReactiveProperty 7.8.3. I updated it to the latest (7.11) but still I get this error.
Any idea? 
Thanks!

**Comment by runceel** [comment]
@nomadev Have you fixed the issue?

**Comment by nomadev** [comment]
Hi @runceel. Thank you very much for asking.
I closed the issue as it could be originated not in ReactiveProperty, but on a DevExpress Grid bound to my collection. 
On the DevExpress library version I were using there was a bug that could trigger the exception ( T976691  – Removing an item from the ColumnsSource collection causes an exception). I updated it to the latest version and, for now, I didn't experience the issue anymore.
